auth/useRegister.ts

```diff
@@ -0,0 +1,38 @@
+import { useState, useMemo } from 'react';
+import firebase from 'firebase/app';
+import { AuthHookType } from '../util';
+
+export type registerHook = AuthHookType<
+  firebase.auth.UserCredential,
+  firebase.FirebaseError
+>;
+
+export default (
+  auth: firebase.auth.Auth,
+  email: string,
```

Comment:
As above, I think it would be better to make `email` and `password` variables that are passed into the returned `register` function